code/datums/elements/swimming_tile.dm

```diff
@@ -70,9 +70,13 @@
 	. = ..()
 	stamina_per_second = ticking_stamina_cost
 	oxygen_per_second = ticking_oxy_damage
+	if (!HAS_TRAIT(owner, TRAIT_SWIMMER))
+		owner.add_movespeed_modifier(/datum/movespeed_modifier/swimming_deep)
 	RegisterSignal(owner, SIGNAL_REMOVETRAIT(TRAIT_IMMERSED), PROC_REF(stop_swimming))
 
 /datum/status_effect/swimming/on_remove()
+	if (!HAS_TRAIT(owner, TRAIT_SWIMMER))
+		owner.remove_movespeed_modifier(/datum/movespeed_modifier/swimming_deep)
```

Comment:
You should safely be able to just always _remove_ the modifier regardless of if they have the trait, spares weird edge cases where someone gains the trait while in water and then doesn't lose the speed modifier.

Reply:
> You should safely be able to just always _remove_ the modifier regardless of if they have the trait, spares weird edge cases where someone gains the trait while in water and then doesn't lose the speed modifier.

k